File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/PIDTuner.java
package org.firstinspires.ftc.teamcode.opmodes;

import com.acmerobotics.dashboard.FtcDashboard;
import com.acmerobotics.dashboard.config.Config;
import com.acmerobotics.dashboard.telemetry.MultipleTelemetry;
import com.qualcomm.robotcore.eventloop.opmode.OpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;

import org.firstinspires.ftc.teamcode.common.subsystems.ArmPID;
import org.firstinspires.ftc.teamcode.common.subsystems.LiftPID;

@Config
@TeleOp
public class PIDTuner extends OpMode {
    public ArmPID arm;
    public LiftPID lift;

    public static int armTarget = 0;
    public static int liftTarget = 0;

    @Override
    public void init() {
        telemetry = new MultipleTelemetry(telemetry, FtcDashboard.getInstance().getTelemetry());
        arm = new ArmPID(hardwareMap);
        lift = new LiftPID(hardwareMap);
    }

    @Override
    public void loop() {
        arm.setTarget(armTarget);
        arm.run();

        lift.setTarget(liftTarget);
        lift.run();

        telemetry.addData("arm pos: ", arm.rightRotate.getCurrentPosition());
        telemetry.addData("arm target: ", armTarget);

        telemetry.addData("lift pos: ", lift.lift.getCurrentPosition());
        telemetry.addData("lift target; ", liftTarget);

    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/PIDTuner.kt
package org.firstinspires.ftc.teamcode.opmodes

import com.acmerobotics.dashboard.FtcDashboard
import com.acmerobotics.dashboard.config.Config
import com.acmerobotics.dashboard.telemetry.MultipleTelemetry
import com.arcrobotics.ftclib.command.CommandOpMode
import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import org.firstinspires.ftc.teamcode.common.subsystems.Arm
import org.firstinspires.ftc.teamcode.common.subsystems.Lift

@Config
@TeleOp
class PIDTuner : CommandOpMode() {
    val telem = MultipleTelemetry(telemetry, FtcDashboard.getInstance().telemetry)

    val arm: Arm by lazy {
        Arm(hardwareMap, telem, lift.lift::getCurrentPosition)
    }
    val lift: Lift by lazy {
        Lift(hardwareMap)
    }

    override fun initialize() {
    }

    override fun run() {
        super.run()
        arm.target = armTarget
        lift.target = liftTarget

        telem.addData("arm pos: ", arm.rightRotate.currentPosition + Arm.starting_ticks)
        telem.addData("arm angle: ", arm.angle)
        telem.addData("arm target: ", armTarget)

        telem.addData("lift pos: ", lift.lift.currentPosition)
        telem.addData("lift target; ", liftTarget)
    }

    companion object {
        @JvmField
        var armTarget: Int = 0

        @JvmField
        var liftTarget: Int = 0
    }
}